bin/cml/runner.js

```diff
@@ -229,59 +253,56 @@ const runLocal = async (opts) => {
     dockerVolumes
   });
 
-  const dataHandler = async (data) => {
-    const log = await cml.parseRunnerLog({ data });
-    log && winston.info('runner status', log);
+  proc.stderr.on('data', dataHandler);
+  proc.stdout.on('data', dataHandler);
+  proc.on('disconnect', () =>
+    shutdown({ ...opts, reason: `runner disconnected` })
+  );
+  proc.on('close', (exit) =>
+    shutdown({ ...opts, reason: `runner closed with exit code ${exit}` })
+  );
 
-    if (log && log.status === 'job_started') {
-      RUNNER_JOBS_RUNNING.push({ id: log.job, date: log.date });
-      RUNNER_TIMEOUT_TIMER = 0;
-    } else if (log && log.status === 'job_ended') {
-      const { job } = log;
-
-      const waitCompletedPipelineJobs = () => {
-        return new Promise((resolve, reject) => {
-          try {
-            if (RUNNER_JOBS_RUNNING.length === 1) {
-              resolve([RUNNER_JOBS_RUNNING[0].id]);
-              return;
-            }
-
-            const watcher = setInterval(async () => {
-              const jobs = (
-                await cml.pipelineJobs({ jobs: RUNNER_JOBS_RUNNING })
-              )
-                .filter((job) => job.status === 'completed')
-                .map((job) => job.id);
-
-              if (jobs.length) {
-                resolve(jobs);
-                clearInterval(watcher);
-              }
-            }, 5 * 1000);
-          } catch (err) {
-            reject(err);
+  RUNNER = proc;
+  ({ id: RUNNER_ID } = await cml.runnerByName({ name }));
+
+  if (parseInt(idleTimeout) > 0) {
+    const watcher = setInterval(async () => {
+      let idle = !RUNNER_JOBS_RUNNING.length;
+
+      try {
+        if (cml.driver === 'github') {
+          const job = await cml.runnerJob({ runnerId: RUNNER_ID });
+
+          if (!job && !idle) {
+            winston.error(
+              `Runner should be idle. Reseting jobs. Will try again in ${idleTimeout} secs`
+            );
```

Comment:
Should be a warning?

Reply:
gave you my first 👍 👎 !!!!!
 initially thought about a warn, but indeed the nature of it is an error, recoverable but an error... not sure

Reply:
I see you're developing a taste for my creative 👍🏼 👎🏼 “yes and no” neutral reaction. 😄 I think the same: as long as it's consistent, it's just a matter of style.